src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -632,6 +632,94 @@ private void CleanUp()
         #endregion IDisposable Members
     }
 
+    /// <summary>
+    /// Implements ConvertTo-CliXml command.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2243:AttributeStringLiteralsShouldParseCorrectly")]
+    [Cmdlet(VerbsData.ConvertTo, "CliXml", HelpUri = "")]
+    public sealed class ConvertToClixmlCommand : PSCmdlet
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets input objects to be converted to CliXml object.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets depth of serialization.
+        /// </summary>
+        [Parameter]
+        [ValidateRange(1, int.MaxValue)]
+        public int Depth { get; set; } = 1;
+
+        #endregion Parameters
+
+        #region Private Members
+
+        private readonly List<PSObject> _inputObjectBuffer = new();
+
+        #endregion Private Members
+
+        #region Overrides
+
+        /// <summary>
+        /// Process record.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            _inputObjectBuffer.Add(InputObject);
```

Comment:
What if InputObject is array/list/collection? What is right behavior in the case? Have we tests for the scenario?

Reply:
I've added some tests in e3c4c4f3d. Behaviour seems to hold up but can add more as we find to get more coverage.

Reply:
My concern is about `$content = ConvertTo-CliXml -InputObject @(1,2,3,4)`